### ISSUE TITLE
dockerd panic

### ISSUE THREAD

**Issue by sqleejan** [moby/moby#34869]
dockerd panic with logs:

2017-09-16 00:29:55.672 info docker[41953]: panic: runtime error: invalid memory address or nil pointer dereference
2017-09-16 00:29:55.672 info docker[41953]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1513ab2]
2017-09-16 00:29:55.672 info docker[41953]: goroutine 36780342 [running]:
2017-09-16 00:29:55.672 info docker[41953]: panic(0x17a2b40, 0x25ec1d0)
2017-09-16 00:29:55.672 info docker[41953]: /usr/local/go/src/runtime/panic.go:531 +0x1cf fp=0xc4266c3eb0 sp=0xc4266c3e18
2017-09-16 00:29:55.672 info docker[41953]: runtime.panicmem()
2017-09-16 00:29:55.672 info docker[41953]: /usr/local/go/src/runtime/panic.go:63 +0x5e fp=0xc4266c3ed0 sp=0xc4266c3eb0
2017-09-16 00:29:55.672 info docker[41953]: runtime.sigpanic()
2017-09-16 00:29:55.672 info docker[41953]: /usr/local/go/src/runtime/signal_unix.go:290 +0x29f fp=0xc4266c3f20 sp=0xc4266c3ed0
2017-09-16 00:29:55.672 info docker[41953]: github.com/docker/docker/daemon/logger/syslog.(*syslogger).Log(0xc421bda1b0, 0xc424d60060, 0x0, 0x0)
2017-09-16 00:29:55.672 info docker[41953]: /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/daemon/logger/syslog/syslog.go:137 +0x82 fp=0xc4266c3f70 sp=0xc4266c3f20

**Steps to reproduce the issue:**
1. this problem occur infrequently

**Output of `docker version`:**

```
Version:      17.06.0
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
```

**Output of `docker info`:**

```
Logging Driver: syslog
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
```
**uname -a**

```
Linux xg-mesos-235 3.10.0-514.26.2.el7.x86_64 #1 SMP Tue Jul 4 15:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by thaJeztah** [comment]
Please provide more information as requested in the [issue-template](https://raw.githubusercontent.com/docker/docker/master/.github/ISSUE_TEMPLATE.md) that was shown when you opened this issue.

- provide the _full_ output of `docker version`
- provide the _full_ output of `docker info`
- provide the exact steps to reproduce. If the steps to reproduce require custom images, or source code, provide a _minimal_ test-case to reproduce. Keep in mind that we don't have access to your private images or source code. Without steps to reproduce, or a reproducible case it's unlikely that your issue can be resolved.

Could you also tell;

- if this was a fresh install or an upgrade from an earlier version
- docker 17.06.2 has been released since; 17.06.1 and 17.06.2 contain a lot of bug fixes discovered after the release of docker 17.06
- when does the panic occur? e.g., is this on startup of the daemon, on reboot of the host, after starting a container (steps to reproduce)\
- are there any log entries in your system log before this happened? (e.g. was the host running out of memory?)

**Comment by sqleejan** [comment]
##docker info##
```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 9
Server Version: 17.06.0
Storage Driver: devicemapper
 Pool Name: docker-253:2-12582921-pool
 Pool Blocksize: 65.54kB
 Base Device Size: 17.18GB
 Backing Filesystem: xfs
 Data file: /dev/vg/docker-data
 Metadata file: /dev/vg/docker-metadata
 Data Space Used: 2.318GB
 Data Space Total: 322.1GB
 Data Space Available: 319.8GB
 Metadata Space Used: 6.234MB
 Metadata Space Total: 5.507GB
 Metadata Space Available: 5.501GB
 Thin Pool Minimum Free Space: 32.21GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.135-RHEL7 (2016-11-16)
Logging Driver: syslog
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 6e23458c129b551d5c9871e5174f6b1b7f6d1170
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-514.26.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 40
Total Memory: 125.6GiB
Name: xxx
ID: FUND:N525:S6YW:K56F:R5EB:BMMC:AXQ6:I35G:H3JC:SMNO:K3XA:UNOI
Docker Root Dir: /data/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

##docker version##
```
Client:
 Version:      17.06.0
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Tue Aug 22 05:50:48 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Tue Aug 22 05:58:55 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by sqleejan** [comment]
@thaJeztah I can't reproduce; It occur infrequently. and System status seem be normal when it occur.

why？

**Comment by cpuguy83** [comment]
This is fixed in 17.06.1, here: https://github.com/docker/docker-ce/commit/bac91bd031a8b5b91747a16c5ff785d2a76685b4

Please upgrade and let us know if this solves it for you.
Closing. Thanks!
